x/contracts/wasmlanche/wasmlanche-test/src/lib.rs

```diff
@@ -0,0 +1,382 @@
+
+                    let serialized_args = memory
+                        .data(&caller)
+                        .get(ptr..ptr + len)
+                        .expect("data should exist");
+
+                    let args: Vec<(StateKey, StateValue)> =
+                        borsh::from_slice(serialized_args).expect("failed to deserialize args");
+
+                    caller.data_mut().set(args);
+                },
+            )
+            .expect("failed to link `state.put` function");
+
+        linker
+            .func_wrap(
+                "state",
+                "get",
+                move |mut caller: Caller<'_, StoreData>, ptr: u32, len: u32| {
+                    let Extern::Memory(memory) = caller
+                        .get_export("memory")
+                        .expect("memory should be exported")
+                    else {
+                        panic!("export `memory` should be of type `Memory`");
+                    };
+
+                    let value = {
+                        let (ptr, len) = (ptr as usize, len as usize);
+
+                        let state_key = memory
+                            .data(&caller)
+                            .get(ptr..ptr + len)
+                            .expect("data should exist");
+
+                        caller.data().get(state_key).cloned()
+                    };
+
+                    let Some(value) = value else {
+                        return 0;
+                    };
+
+                    let Some(Extern::Func(f)) = caller.get_export(ALLOC_FN_NAME) else {
+                        panic!("export `{ALLOC_FN_NAME}` should exist");
+                    };
+
+                    // TODO:
+                    // should able to call this unchecked as it is checked
+                    // immediately after instantiation of any instance
+                    let alloc = f
+                        .typed::<AllocParam, AllocReturn>(&mut caller)
+                        .expect("allocate function should always exist");
+
+                    let offset = alloc
+                        .call(&mut caller, value.len() as u32)
+                        .expect("failed to allocate memory");
+
+                    memory
+                        .write(&mut caller, offset as usize, &value)
+                        .expect("failed to write value to memory");
+
+                    offset
+                },
+            )
+            .expect("failed to link `state.get` function");
+
+        Self {
+            engine,
+            module,
+            linker,
+        }
+    }
+
+    #[inline(always)]
+    pub fn build(&self) -> TestCrate {
+        let Self {
+            engine,
+            module,
+            linker,
+        } = self;
+
+        let store_data = StoreData {
+            call_result: None,
+            state: HashMap::default(),
+            limiter: StoreLimitsBuilder::new().memory_size(18 * 0x10000).build(),
```

Comment:
why 18 * 65536 bytes? If no particular reason can we make this a more human readable calculation such as
180 * 1000 = 180 KB

Reply:
1 << 16 is the default page-size. I can leave a comment